lib/dor/workflow/client/status.rb

```diff
@@ -101,24 +82,33 @@ def simplified_status_code(display)
         end
 
         def current_milestones
-          current = []
-          milestones.each do |m|
-            next unless STEPS.key?(m[:milestone]) # milestone name must be in list of known steps
-
+          milestones
+            # milestone name must be in list of known steps
+            .select { |m| STEPS.key?(m[:milestone]) }
             # Two possible ways the version can indicate the milestone is part of the current version:
             # if m[:version] is nil, then the milestone is active (version 0 becoming version 1)
             # if m[:version] is matches the current version, then the milestone is archived with the current version
-            next unless m[:version].nil? || m[:version] == version
+            .select { |m| m[:version].nil? || m[:version] == version }
+            # registered milestone is only valid for v1
+            .reject { |m| m[:milestone] == 'registered' && version.to_i > 1 }
```

Comment:
Another focus of the remainder of this commit is on smarter use of Ruby iterator methods. I find that using `#each` can conceal intention, and this is a good example. We're using `#each` here as a way to filter a list into shorter list. To reveal that intention more clearly, refactor this as a set of selects and rejects.